agent/lib/kontena/rpc_server.rb

```diff
@@ -28,27 +29,33 @@ def initialize(code, message, backtrace = nil)
     ##
     # @param [Array] message msgpack-rpc request array
     # @return [Array]
-    def handle_request(message)
+    def handle_request(ws_client, message)
       msg_id = message[1]
       handler = message[2].split('/')[1]
       method = message[2].split('/')[2]
       if klass = HANDLERS[handler]
         begin
           debug "rpc request: #{klass.name}##{method} #{message[3]}"
           result = klass.new.send(method, *message[3])
-          return [1, msg_id, nil, result]
+          send_message(ws_client, [1, msg_id, nil, result])
         rescue RpcServer::Error => exc
-          return [1, msg_id, {code: exc.code, message: exc.message, backtrace: exc.backtrace}, nil]
+          send_message(ws_client, [1, msg_id, {code: exc.code, message: exc.message, backtrace: exc.backtrace}, nil])
         rescue => exc
           error "#{exc.class.name}: #{exc.message}"
```

Comment:
If we're going to change the RPC handlers to raise errors instead of logging them, then this should probably also log the RPC request and parameters for text.